EpiAware/test/predictive_checking/fast_approx_for_r.jl

```diff
@@ -0,0 +1,73 @@
+#=
+# Fast approximation for `r` from `R₀`
+I use the negative moment generating function (MGF).
+
+Let
+```math
+G(r) = \sum_{i=1}^{\infty} w_i e^{-r i}.
+```
+and
+```math
+f(r, \mathcal{R}_0) = \mathcal{R}_0 G(r) - 1.
+```
+then the connection between `R₀` and `r` is given by
+```math
+f(r, \mathcal{R}_0) = 0.
+```
+Given an estimate of $\mathcal{R}_0$ we implicit solve for $r$ using a root
+finder algorithm. In this note, I test a fast approximation for $r$ which
+should have good autodifferentiation properties. The idea is to start from the
```

Comment:
you say should here and you say it does in the comment on Dylan's PR but I see not testing evidence?

Reply:
The test script is [here](https://github.com/CDCgov/Rt-without-renewal/blob/40-initialisation-infection-generation-processes/EpiAware/test/predictive_checking/fast_approx_for_r.jl).

I'm not sure how much testing this needs? Its a solvable small $r$ approximation to start a small number of Newton method steps. The main failure mode is incorrectly writing the adjoint method for the PGF of the generation distribution which you can diagnose visually (because it will diverge or go to wrong value fast).

Reply:
I've updated the doc to the `R_to_r` function.

Reply:
my complaint here is specifically about evidence that "it works well under AD" but I am happy that it probable does (can see little reason it wouldn't)

Reply:
Ah right! OK that is based on more general principles, e.g. that this function pre-defines the number of iterations it will do rather than basing it on abstol etc.

Reply:
got to explain more Sam for us slow folk :)